hack/setup-cluster-wide-proxy.sh

```diff
@@ -0,0 +1,488 @@
+# create the proxy stack and then get its IP
+PROXY_URI="s3://${PROXY_NAME}/proxy.ign"
+
+# push the generated ignition to the s3 bucket
+aws --region "${REGION}" s3 cp ${TEMP_DIR}/proxy.ign "${PROXY_URI}"
+echo "Uploaded ${TEMP_DIR}/proxy.ign to S3 bucket ${PROXY_NAME} in region ${REGION}"
+
+aws --region "${REGION}" s3api put-object-acl --bucket "${PROXY_NAME}" --key "proxy.ign" --acl public-read #--acl bucket-owner-full-control
+echo "Updating S3 bucket object ${PROXY_URI}: put-object-acl succeeded"
+
+generate_proxy_aws_template
+
+EXPIRATION_DATE=$(date -d '4 hours' --iso=minutes --utc)
+TAGS="Key=expirationDate,Value=${EXPIRATION_DATE}"
+
+aws --region "${REGION}" cloudformation create-stack \
+  --stack-name "${PROXY_NAME}" \
+  --template-body "$(cat "${TEMP_DIR}/cluster_proxy_aws_template.yaml")" \
+  --tags "${TAGS}" \
+  --capabilities CAPABILITY_NAMED_IAM \
+  --parameters \
+  ParameterKey=ClusterName,ParameterValue="${PROXY_NAME}" \
+  ParameterKey=VpcId,ParameterValue="${VPC_ID}" \
+  ParameterKey=ProxyIgnitionLocation,ParameterValue="${PROXY_URI}" \
+  ParameterKey=InfrastructureName,ParameterValue="${INFRASTRUCTURE_NAME}" \
+  ParameterKey=Ami,ParameterValue="${AMI}" \
+  ParameterKey=PublicSubnet,ParameterValue="${PUBLIC_SUBNET_ID}" &
+
+wait "$!"
+echo "Created stack ${PROXY_NAME}"
+
+aws --region "${REGION}" cloudformation wait stack-create-complete --stack-name "${PROXY_NAME}" &
+wait "$!"
+echo "Waited for stack"
+
+echo ""
+INSTANCE_ID=$(aws cloudformation describe-stacks --stack-name "${PROXY_NAME}" --query 'Stacks[].Outputs[?OutputKey == `ProxyId`].OutputValue' --region "${REGION}"  --output text)
+echo "Created proxy EC2 intance with id ${INSTANCE_ID}" 
+
+echo ""
+echo "Fetching IPs for proxy intance ${PROXY_NAME}"
+PRIVATE_PROXY_IP=$(aws  cloudformation describe-stacks --stack-name "${PROXY_NAME}" --query 'Stacks[].Outputs[?OutputKey == `ProxyPrivateIp`].OutputValue' --region "${REGION}" --output text)
+PUBLIC_PROXY_IP=$(aws cloudformation describe-stacks --stack-name "${PROXY_NAME}" --query 'Stacks[].Outputs[?OutputKey == `ProxyPublicIp`].OutputValue' --region "${REGION}" --output text)
+echo "PRIVATE_PROXY_IP: ${PRIVATE_PROXY_IP}"
+echo "PUBLIC_PROXY_IP: ${PUBLIC_PROXY_IP}"
+
+echo ""
+echo "Generating URL for proxy ${PROXY_NAME}"
+PROXY_URL="http://${PROXY_NAME}:${PASSWORD}@${PRIVATE_PROXY_IP}:3128"
+# due to https://bugzilla.redhat.com/show_bug.cgi?id=1750650 we don't use a tls end point for squid
+
+echo ""
+echo "Patching proxy specs httpProxy and httpsProxy with URL: ${PROXY_URL}"
+oc patch proxy/cluster --type=merge -p '{"spec":{"httpProxy":"'"${PROXY_URL}"'","httpsProxy":"'"${PROXY_URL}"'","noProxy":".example.com"}}'
+
+echo ""
+echo "Fetching proxy specs" 
+oc get proxy cluster -o yaml
+
+echo ""
+echo "Cleanup"
+echo ""
+echo "To delete proxy cloudformation"
+echo "    aws cloudformation delete-stack --stack-name ${PROXY_NAME} --region ${REGION}"
```

Comment:
for each of these steps, maybe we should make the instruction line a bash comment:
`echo "# To delete proxy cloudformation"`
That way all the cleanup steps can be copy pasted together rather than one by one

Reply:
The goal here is to provide supporting information to ease the cleanup steps, not to perform the cleanup step. I would not recommend a bulk execution of the steps. 

In case the  cleanup process becomes more complex, this script can generated a sort of post-script command to clean everything. As of now, it is just one command to remove the cloudformation.